Upgrade to 2.0 (#323)
Sylius/InvoicingPlugin: UPGRADE.md

> ### UPGRADE FROM 0.24.X TO 1.0.0

Reviewer: We will probably release support for Sylius 2.0 in version 2.0 of this plugin, and release earlier 1.0, but we could adjust it later 🖖🏻

1. Support for Sylius 2.0 has been added, it is now the recommended Sylius version to use with InvoicingPlugin.
2. Support for Sylius 1.13 and 1.12 has been dropped, upgrade your application to [Sylius 2.0](https://github.com/Sylius/Sylius/blob/2.0/UPGRADE-2.0.md).

> 3. Support for PHP has been increased to 8.2

Reviewer: ```suggestion
3. The minimum supported PHP version has been changed to 8.2
```
🤔
